app/models/page.rb

```diff
@@ -2,6 +2,18 @@ class Page < ApplicationRecord
   include UuidPrimaryKey
 
   has_many :versions, -> { order(capture_time: :desc) }, foreign_key: 'page_uuid', inverse_of: :page
+  has_one :latest,
+    (lambda do
+      # DISTINCT ON requires the first ORDER to be the distinct column(s)
+      relation = order('versions.page_uuid')
+      # HACK: This is not public API, but I couldn't find a better way. The
```

Comment:
Reaching below the public API every now and then is a good sign that you're doing something interesting.